mmedit/models/backbones/encoder_decoders/necks/aot_neck.py

```diff
@@ -0,0 +1,188 @@
+import torch
+import torch.nn as nn
+from mmcv.cnn import ConvModule
+
+from mmedit.models.registry import COMPONENTS
+
+
+@COMPONENTS.register_module()
+class AOTBlockNeck(nn.Module):
+    """Dilation Backbone used in Global&Local model.
+
+    This implementation follows:
+    Globally and locally Consistent Image Completion
+
+    Args:
+        in_channels (int): Channel number of input feature.
+        conv_type (str): The type of conv module. In DeepFillv1 model, the
+            `conv_type` should be 'conv'. In DeepFillv2 model, the `conv_type`
+            should be 'gated_conv'.
+        norm_cfg (dict): Config dict to build norm layer.
+        act_cfg (dict): Config dict for activation layer, "relu" by default.
+        kwargs (keyword arguments).
+    """
+
+    def __init__(self,
+                 in_channels=256,
+                 act_cfg=dict(type='ReLU'),
+                 dilation_rates='1+2+4+8',
+                 num_aotblock=8,
+                 **kwargs):
+        super().__init__()
+
+        self.dilation_rates = list(map(int, list(dilation_rates.split('+'))))
+        
+        
+        self.num_aotblock = num_aotblock
+
+        self.middle = nn.Sequential(*[AOTBlock(256, self.dilation_rates) for _ in range(num_aotblock)])
+        
+        '''
+        
+        self.model = nn.Sequential()
+        
+        for i in range(self.num_aotblock):
+            self.model.add_module(
+                f"aotblock{i + 1}", 
+                AOTBlock(
+                    in_channels=in_channels,
+                    dilation_rates=self.dilation_rates,
+                    act_cfg=act_cfg,
+                )
+            )
+        '''
+        
+        
+        
+    def forward(self, x):
+        x = self.middle(x)
+        return x
+
+
+class AOTBlock(nn.Module):
+    def __init__(self, dim, rates):
+        super(AOTBlock, self).__init__()
+        self.rates = rates
+        for i, rate in enumerate(rates):
+            self.__setattr__(
+                'block{}'.format(str(i).zfill(2)), 
+                nn.Sequential(
+                    nn.ReflectionPad2d(rate),
+                    nn.Conv2d(dim, dim//4, 3, padding=0, dilation=rate),
+                    nn.ReLU(True)))
+        self.fuse = nn.Sequential(
+            nn.ReflectionPad2d(1),
+            nn.Conv2d(dim, dim, 3, padding=0, dilation=1))
+        self.gate = nn.Sequential(
+            nn.ReflectionPad2d(1),
+            nn.Conv2d(dim, dim, 3, padding=0, dilation=1))
+
+    def forward(self, x):
+        out = [self.__getattr__(f'block{str(i).zfill(2)}')(x) for i in range(len(self.rates))]
+        out = torch.cat(out, 1)
+        out = self.fuse(out)
+        mask = my_layer_norm(self.gate(x))
+        mask = torch.sigmoid(mask)
+        return x * (1 - mask) + out * mask
+
+
+def my_layer_norm(feat):
```

Comment:
Is this a special operation, different from the conventional layer normalization?